sklearn/impute.py

```diff
@@ -226,7 +226,13 @@ def fit(self, X, y=None):
                              "numerical value when imputing numerical "
                              "data".format(fill_value))
 
-        if sparse.issparse(X):
+        if sparse.issparse(X) and self.missing_values == 0:
```

Comment:
Nitpick: I would avoid testing twice "issparse": I would test once issparse, and add a second test inside this if statement.